### PR TITLE
Fullscreen support #38

### PR DESCRIPTION
**Pull request by PoisonousJohn**
Added support for fullscreen:

- Going fullscreen if html element requested fullscreen
- Shortcuts to go fullscreen F11/ Ctrl+Meta+F
- Hiding of a toolbar during fullscreen mode

### CODE REVIEW

File: src/ui/window/BrowserWindow.qml [hunk 1]
```diff
@@ -185,6 +186,13 @@ ApplicationWindow {
                 ]
                 rightActions: [
                     Action {
+                        iconName: window.isFullScreen ? "navigation/fullscreen_exit" : "navigation/fullscreen"
+                        onTriggered: {
+                            toggleFullScreen()
+                        }
```
Comment (timsueberkrueb) [inline src/ui/window/BrowserWindow.qml]: Can we move this to the right of the refresh button?
Also can we hide this button in normal mode, showing it only in fullscreen mode?
I think we should also have an item in the menu (which is always shown).
That way we are consistent with the download action which has a menu item but is only shown in the toolbar when there is an active download.

**Comment by timsueberkrueb** [comment]
Hey, thank you very much! Looking good :+1: I really like the show/hide animation.
Only suggestion is about the toolbar button, see my comment.

File: src/ui/window/BrowserWindow.qml [hunk 2]
```diff
@@ -419,6 +420,16 @@ ApplicationWindow {
         }
 
         MenuItem {
+            text: "Fullscreen"
+            iconName: "navigation/fullscreen"
+            visible: !window.isFullScreen
```
Comment (timsueberkrueb) [inline src/ui/window/BrowserWindow.qml]: @JohnPoison can we show this always?

Comment (PoisonousJohn) [reply src/ui/window/BrowserWindow.qml]: I guess

File: src/ui/window/BrowserWindow.qml [hunk 3]
```diff
@@ -426,9 +426,7 @@ ApplicationWindow {
 
         MenuItem {
             text: "Fullscreen"
-            iconName: "navigation/fullscreen"
-            visible: !window.isFullScreen
-            height: visible ? undefined : 0
+            iconName: !window.isFullScreen ? "navigation/fullscreen" : "navigation/fullscreen_exit"
```
Comment (timsueberkrueb) [inline src/ui/window/BrowserWindow.qml]: :+1:

Comment (timsueberkrueb) [reply src/ui/window/BrowserWindow.qml]: @JohnPoison maybe `text: window.isFullScreen? "Show normal" : "Fullscreen"` too? What do you think?

**Review by timsueberkrueb — APPROVED** [review]
Great, thank you!

**Pull request merged**